python-backend/app/api/mines/variances/models/variance.py

```diff
@@ -0,0 +1,60 @@
+from datetime import datetime
+
+from sqlalchemy.dialects.postgresql import UUID
+from sqlalchemy.schema import FetchedValue
+
+from ....utils.models_mixins import AuditMixin, Base
+from app.extensions import db
+
+
+class Variance(AuditMixin, Base):
+    __tablename__ = "variance"
+    variance_id = db.Column(db.Integer, primary_key=True, server_default=FetchedValue())
+    compliance_article_id = db.Column(db.Integer,
+                                      db.ForeignKey('compliance_article.compliance_article_id'),
+                                      server_default=FetchedValue())
+    mine_guid = db.Column(UUID(as_uuid=True), db.ForeignKey('mine.mine_guid'))
+    note = db.Column(db.String(300), server_default=FetchedValue())
+    issue_date = db.Column(db.DateTime, nullable=False, default=datetime.utcnow)
+    received_date = db.Column(db.DateTime, nullable=False, default=datetime.utcnow)
+    expiry_date = db.Column(db.DateTime,
+                            nullable=False,
+                            default=datetime.strptime('9999-12-31', '%Y-%m-%d'))
+
+    related_documents = db.relationship(
+        'VarianceDocument',
+        backref="variance",
+        lazy='joined')
```

Comment:
`lazy="joined"` means it is lazy loaded, given how tightly the documents are used with the variance, we might get preformance gains out of `lazy="select` meaning it's joined on select.

Reply:
Other way around :)

Reply:
https://docs.sqlalchemy.org/en/latest/orm/relationship_api.html#sqlalchemy.orm.relationship.params.lazy

Reply:
well never mind then.....